### ISSUE TITLE
[flang][openmp] Incorrect diagnostic on Openmp Atomic update statement

### ISSUE THREAD

**Issue by DanielCChen** [llvm/llvm-project#138748]
Consider the following test case
```
    integer (kind = 4) :: a
    real :: c
    a = 1
    c = 3
    d = 4
!$omp atomic
    a = a + c + d
end
```

Flang now issues an error
```
./t.f:7:5: error: Atomic update statement should be of form `a = a operator expr` OR `a = expr operator a`
      a = a + c + d
      ^
```

The code is conforming to the standard.

**Comment by eugeneepshteyn** [comment]
Looking at OpenMP 5.2 spec, section "4.3.1.3 OpenMP Atomic Structured Blocks", page 81, lines 10-13. The following forms are allowed:
```
x = x operator expr
x = expr operator x
```
`expr` here is a scalar expression, and in the reproducer `a = a + c + d`, where `c + d` is a scalar expression. So yes, I agree that this should not result in an error.

**Comment by eugeneepshteyn** [comment]
In `OmpStructureChecker::IsOperatorValid()`, the code seems to only support cases like `a = a + b` and `a = b + a`. Will need to update with more extensive scalar expression support.

**Comment by eugeneepshteyn** [comment]
(Echoing Peter on the other OpenMP issue)

Kiran, please reassign to whomever takes care of OpenMP parsing.
